core/src/main/java/com/linecorp/armeria/client/WebClient.java

```diff
@@ -312,15 +352,50 @@ static WebClientBuilder builder(URI uri) {
         return new WebClientBuilder(uri);
     }
 
+    /**
+     * Returns a new {@link WebClientBuilder} created with the specified {@code protocol}
+     * and base {@link EndpointGroup}.
+     *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    static WebClientBuilder builder(String protocol, EndpointGroup endpointGroup) {
+        return builder(SessionProtocol.of(requireNonNull(protocol, "protocol")), endpointGroup);
+    }
+
     /**
      * Returns a new {@link WebClientBuilder} created with the specified {@link SessionProtocol}
      * and base {@link EndpointGroup}.
      *
+     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
+     *                                  in {@link SessionProtocol}
```

Comment:
nit: I don't think all `SessionProtocal` is supported. 
How about: 
```suggestion
     * @throws IllegalArgumentException if the {@code protocol} is not one of the fields
     *                                  in {@link SessionProtocol#httpValues()} or {@link SessionProtocol#httpsValues()}

```